src/main/scala/scala/tools/refactoring/implementations/OrganizeImports.scala

```diff
@@ -559,3 +565,79 @@ abstract class OrganizeImports extends MultiStageRefactoring with TreeFactory wi
     }.transform(tree)
   }
 }
+
+class OrganizeImportsSpacer(val oiInstance: OrganizeImports) {
+  import oiInstance.global._
+  import scala.tools.nsc.interactive.Global
+
+  private def topMostClassOrModule(stats: List[Tree]): Option[ImplDef] = stats.find {
+    case classDefOrModuleDef: ImplDef => true
+    case _ => false
+  }.asInstanceOf[Option[ImplDef]]
+
+  private def linesDiffInOrigFile(classOrModule: Option[ImplDef], importt: Option[Import]): Option[Int] =
+    for {
+      cm <- classOrModule
+      imp <- importt
+      sourceFile = cm.pos.source
+      cmLine = sourceFile.offsetToLine(cm.pos.start)
+      impLine = sourceFile.offsetToLine(imp.pos.end)
+    } yield cmLine - impLine
+
+  private def lastImportInPackageScope(stats: List[Tree]): Option[Import] = topMostClassOrModule(stats).flatMap { topMost =>
+    val index = stats.indexOf(topMost)
+    stats.take(index).reverse.dropWhile { !_.isInstanceOf[Import] }.headOption.asInstanceOf[Option[Import]]
+  }
+
+  object AddSpacer {
+    import scala.tools.refactoring.common.TextChange
+
+    private def areNeighborLinesInOrigFile(classOrModule: Option[ImplDef], importt: Option[Import]): Boolean =
+      linesDiffInOrigFile(classOrModule, importt).map { _ == 1 }.getOrElse(false)
+
+    def apply(tree: Global#Tree): List[Change] = tree match {
+      case pkgDef @ PackageDef(_, stats) =>
+        val cm = topMostClassOrModule(stats)
+        val limp = lastImportInPackageScope(stats)
+        if (areNeighborLinesInOrigFile(cm, limp)) {
+          val pcm = cm.get
+          val sourceFile = pcm.pos.source
+          val from = pcm.pos.start - pcm.pos.lineContent.takeWhile { _.isWhitespace }.size
+          List(TextChange(sourceFile, from, from, Properties.lineSeparator))
+        } else Nil
+    }
+  }
+
+  object ReduceSpacer {
+    import scala.tools.refactoring.common.TextChange
+
+    private def isNeededToReduceEmptyLines(classOrModule: Option[ImplDef], importt: Option[Import], stats: List[Tree]): Boolean = {
```

Comment:
Instead of having options as arguments, you could unpack them already in the caller function and be done with it in the subfunctions.